virtual-display-driver-cli/src/client.rs

```diff
@@ -1,110 +1,139 @@
-use std::io::Write as _;
+use std::{
+    collections::{BTreeSet, HashSet},
+    io::Write as _,
+};
 
 use driver_ipc::Monitor;
-use eyre::Context;
+use eyre::Context as _;
+use joinery::Joinable as _;
 use win_pipes::{NamedPipeClientReader, NamedPipeClientWriter};
 
 pub struct Client {
-    reader: NamedPipeClientReader,
     writer: NamedPipeClientWriter,
+    state: Vec<Monitor>,
 }
 
 impl Client {
     pub fn connect() -> eyre::Result<Self> {
-        let (reader, writer) = win_pipes::NamedPipeClientOptions::new("virtualdisplaydriver")
-            .wait()
-            .access_duplex()
-            .mode_message()
-            .create()?;
-
-        Ok(Self { reader, writer })
+        let (mut reader, mut writer) =
+            win_pipes::NamedPipeClientOptions::new("virtualdisplaydriver")
+                .wait()
+                .access_duplex()
+                .mode_message()
+                .create()
+                .context("Failed to connect to Virtual Display Driver; please ensure the driver is installed and working. Other program using the driver must also be closed, such as the Virtual Display Driver Control app.")?;
```

Comment:
Here's what this error looks like now:

![Terminal output with the following message: "Error: Failed to connect to Virtual Display Driver; please ensure the driver is installed and working. Other program using the driver must also be closed, such as the Virtual Display Driver Control app". Following that is a "Caused by" message with a Windows I/O error message and a stacktrace.](https://github.com/MolotovCherry/virtual-display-rs/assets/1362179/25f2261f-5aee-470f-87c6-722091a4b9f9)

Originally, I thought about doing this as a custom error type that gets threaded through to the top-level `main` function. I ultimately thought that just using [`.context()`](https://docs.rs/eyre/0.6.11/eyre/trait.WrapErr.html#tymethod.context) to wrap the existing error would be better for a number of reasons. It was an easier change to make, it has nice formatting built in, and it prints a full stack trace with lower-level error messages as well (which I don't see as a hard requirement, but is a nice bonus since the CLI is a tool intended more for technically-oriented folks). Replicating all that with a custom error type would be doable, but would take a bit of work.

Reply:
`context` suffices just fine! It's always useful to see the more technical error in case it's something else.